avro-fastserde/src/main/java/com/linkedin/avro/fastserde/PrimitiveFloatList.java

```diff
@@ -60,44 +76,63 @@ public PrimitiveFloatList(Collection<Float> c) {
    * @throws IOException on io errors
    */
   public static Object readPrimitiveFloatArray(Object old, Decoder in) throws IOException {
-    long l = in.readArrayStart();
-    if (l > 0) {
-      PrimitiveFloatList array = (PrimitiveFloatList) newPrimitiveFloatArray(old, (int) l);
+    long length = in.readArrayStart();
+    long totalLength = 0;
+
+    if (length > 0) {
+      PrimitiveFloatList array = (PrimitiveFloatList) newPrimitiveFloatArray(old);
+
       do {
-        for (long i = 0; i < l; i++) {
-          array.addPrimitive(in.readFloat());
-        }
-        l = in.arrayNext();
-      } while (l > 0);
+        // Allocate ByeBuffer of size array length * Float.BYTES to hold the array values
+        ByteBuffer byteBuffer = ByteBuffer.allocate((int)length * FLOAT_SIZE).order(ByteOrder.LITTLE_ENDIAN);
+        in.readFixed(byteBuffer.array(), 0, (int)(length * FLOAT_SIZE));
```

Comment:
This will allocate a new `byte[]` inside the `ByteBuffer` and copy the content from the `Decoder` into it, right? Can we manage to get hold of the original `byte[]` that was passed to the `Decoder` instead, and have that one back the `ByteBuffer`, without any copy?

Reply:
wish I could do that. BinaryDecoder does not give us access for that. Probably better approach would to fork BinaryDecoder which would give access to the raw bytes.